sw/device/tests/sram_ctrl_execution_test.c

```diff
@@ -0,0 +1,44 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/dif/dif_sram_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static dif_sram_ctrl_t sram_ctrl;
+
+enum {
+  kSramStart = TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR,
+  kSramEnd = TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR +
+             TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_SIZE_BYTES,
+};
+
+OT_SECTION(".data.sram_function_test")
+OT_NOINLINE()
+static void sram_function_test(void) {
+  uint32_t pc = 0;
+  asm volatile("auipc %[pc], 0;" : [pc] "=r"(pc));
+  LOG_INFO("PC: %p, SRAM: [%p, %p)", pc, kSramStart, kSramEnd);
+  CHECK(pc >= kSramStart && pc < kSramEnd, "PC is outside the main SRAM");
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_sram_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_MAIN_REGS_BASE_ADDR),
+      &sram_ctrl));
+  CHECK_DIF_OK(dif_sram_ctrl_exec_set_enabled(&sram_ctrl, kDifToggleEnabled));
```

Comment:
As far as I understood the original test was attempting to execute in the sram with that feature disabled. Which generates an exception that was handled and verified.

Reply:
Can you take another look? The original test also only checks the success case but in a more convoluted way: attempts to execute the start of ret ram, expects an exception, and checks that exception has occurred.

Reply:
I agree what you say about the original: "attempts to execute the start of ret ram, expects an exception, and checks that exception has occurred", but the new test doesn't test the exception case, right?

Reply:
I see what you mean, yes the new version doesn't test that an exception has occurred because it doesn't have to. We place the function in the main SRAM and check the PC. I'm more than happy to discuss this further offline if you want. Please feel free to ping me over slack when you get the chance.

Reply:
Discussed this offline with @engdoreis and I realized that I kept reading the check wrong. The original test was testing the negative case so I'll add that to this PR to keep parity. Thanks for bearing with me @engdoreis!

Reply:
Done. I was able to add a negative test for the retention SRAM and since execution from the main SRAM is enabled unconditionally in the RMA life cycle state. @engdoreis PTAL.